src/rootPages/Designer/ui_work_interface_list_newPage.js

```diff
@@ -0,0 +1,271 @@
+         this.currentApplication = application; // remember our current Application.
+         this.BlankTab.applicationLoad(application); // send so parent pagelist can be made
+         //  this.QuickTab.applicationLoad(application);
+      }
+
+      /**
+       * @function hide()
+       *
+       * remove the busy indicator from the form.
+       */
+      hide() {
+         if (this.$component) this.$component.hide();
+      }
+
+      /**
+       * Show the busy indicator
+       */
+      busy() {
+         if (this.$component) {
+            this.$component.showProgress();
+         }
+      }
+
+      /**
+       * Hide the busy indicator
+       */
+      ready() {
+         if (this.$component) {
+            this.$component.hideProgress();
+         }
+      }
+
+      /**
+       * @method done()
+       * Finished saving, so hide the popup and clean up.
+       * @param {interface} obj
+       */
+      done(obj) {
+         this.ready();
+         this.hide(); // hide our popup
+         this.emit("save", obj, this.selectNew);
+         // _logic.callbacks.onDone(null, obj, selectNew, callback); // tell parent component we're done
+      }
+
+      /**
+       * @method save
+       * take the data gathered by our child creation tabs, and
+       * add it to our current application.
+       * @param {obj} values  key=>value hash of model values.
+       * @param {string}  tabKey
+       *        the "key" of the tab initiating the save.
+       * @return {Promise}
+       */
+      async save(values, tabKey) {
+         // must have an application set.
+         if (!this.currentApplication) {
+            webix.alert({
+               title: L("Shoot!"),
+               test: L("No Application Set!  Why?"),
+            });
+            this[tabKey].emit("save.error", true);
+            return false;
+         }
+
+         if (!values) {
+            // SaveButton.enable();
+            // CurrentEditor.formReady();
+            return;
+         }
+
+         // create a new (unsaved) instance of our interface:
+         // this interface only creates Root Pages, or pages related to
+         var newInterface = null;
+         if (values.useParent && values.parent) {
+            // ?????????????????
+            newInterface = values.parent;
+         } else if (values.parent) {
+            newInterface = values.parent.pageNew(values);
+         } else {
+            //page = CurrentApplication.pageNew(values);
+            newInterface = this.currentApplication.pageNew(values);
+         }
+         //
+
+         // have newInterface validate it's values.
+         // if this item supports isValid()
+         if (newInterface.isValid) {
+           var validator = newInterface.isValid();
+           if (validator.fail()) {
+             // cb(validator); // tell current Tab component the errors
+             this[tabKey].emit("save.error", validator);
+             newInterface.label = state.old;
```

Comment:
what's `state`?

Reply:
don't know, this is for the auto-generated page popup. Since it returns immediately and does nothing with that label (as far as I can tell) I removed it